md/barostat/utils.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+import networkx as nx
+from scipy.spatial.distance import pdist
+from typing import List
+from timemachine.lib.potentials import HarmonicBond
+
+
+def compute_box_volume(box: np.ndarray) -> float:
+    assert box.shape == (3, 3)
+    return np.linalg.det(box)
+
+
+def compute_box_center(box: np.ndarray) -> np.ndarray:
+    # assume axis-aligned box (nothing off diagonal)
+    assert box.shape == (3, 3)
+    assert np.linalg.norm(box - np.diag(np.diag(box))) == 0
+
+    center = np.sum(box / 2, axis=0)
+
+    assert center.shape == (3,)
+
+    return center
+
+
+def get_group_indices(harmonic_bond_potential: HarmonicBond) -> List[np.array]:
```

Comment:
Note that in certain cases (eg. core restraints for ABFE), we will actually have two harmonic bond terms, one for the regular valences, and one for the restraining term. It's probably more flexible to just modify the input of this function to either

1) a list of HarmonicBonds
or perhaps slightly cleaner:
2) have it take in a simple Bx2 list of idxs (and leave it up to the caller to pass merge the indices).

Reply:
Updated the signature as suggested here https://github.com/proteneer/timemachine/pull/433/commits/954937ce6ba2a2b4c44a2c6d36c5efc59c9318e7 . Later, might be worthwhile to remove both of these and replace with an "alchemical topology" object or similar, that helps keep track of what bonds are present regardless of whether they're represented as harmonic bond terms, distance constraints, etc. down the road.